waspc/src/Wasp/AppSpec.hs

```diff
@@ -14,5 +16,11 @@ data AppSpec = AppSpec
   { -- | List of declarations like App, Page, Route, ... that describe the web app.
     decls :: [Decl],
     -- | List of external code files (they are referenced/used by the declarations).
-    externalCodeFiles :: ExternalCode.File
+    externalCodeFiles :: [ExternalCode.File],
+    -- | Absolute path to the directory in wasp project source that contains external code files.
+    externalCodeDirPath :: !(Path' Abs (Dir ExternalCode.SourceExternalCodeDir)),
+    -- | Absolute path to the directory in wasp project source that contains database migrations.
+    migrationsDir :: Maybe (Path' Abs (Dir DbMigrationsDir)),
+    dotEnvFile :: Maybe (Path' Abs File'),
+    isBuild :: Bool
```

Comment:
shall we add comment/explanation for these two properties as well, since all others have one? Or is it already specified somewhere else

Reply:
There wasn't any before so I didn't add it now, but I could! I will add it in this new PR where I am merging AppSpec with the rest of the codebase.